src/avx512-16bit-common.h

```diff
@@ -316,10 +316,53 @@ static void qselect_16bit_(type_t *arr,
     type_t biggest = vtype::type_min();
     int64_t pivot_index = partition_avx512<vtype>(
             arr, left, right + 1, pivot, &smallest, &biggest);
-    if ((pivot != smallest) && (pos < pivot_index))
-        qselect_16bit_<vtype>(arr, pos, left, pivot_index - 1, max_iters - 1);
-    else if ((pivot != biggest) && (pos >= pivot_index))
-        qselect_16bit_<vtype>(arr, pos, pivot_index, right, max_iters - 1);
+
+    if (pos < pivot_index) {
```

Comment:
Would prefer to keep the if conditions the way they are, just more readable I think.